addons/web_editor/static/lib/odoo-editor/test/spec/format.test.js

```diff
@@ -961,6 +961,13 @@ describe('setTagName', () => {
                 contentAfter: '<h1 class="text-uppercase">[abcd]</h1>',
             });
         });
+        it('should turn a inline elements within a div into a heading 1', async () => {
+            await testEditor(BasicEditor, {
+                contentBefore: '<div><font style="color: red;">a[b</font><span style="font-size: 12px;">c]d</span><font style="color: red;">ef</font><p>gh</p><font style="color: red;">ij</font><span style="font-size: 12px;">kl</span></div>',
+                stepFunction: editor => editor.execCommand('setTag', 'h1'),
+                contentAfter: '<div><h1><font style="color: red;">a[b</font><span style="font-size: 12px;">c]d</span><font style="color: red;">ef</font></h1><p>gh</p><font style="color: red;">ij</font><span style="font-size: 12px;">kl</span></div>',
```

Comment:
The span that contains the text "ef" was not selected so it should definitely not become a heading along with the others before it.